src/account/components/AccountLogin.js

```diff
@@ -44,25 +44,45 @@ const GoogleIcon = props => {
   return <SvgIcon component={GoogleSvg} {...props} />;
 };
 
-const appendReferrer = (url, referrer) =>
-  referrer ? `${url}&state=${referrer}` : url;
+const appendReferrer = (url, referrer) => {
+  return referrer
+    ? `${url}&state=account?referrerBase64=${btoa(referrer)}`
+    : url;
+};
```

Comment:
issue: this URL is malformatted. You can't have two `?`s. Please use a URL parsing function to add the arguments.

Reply:
Perhaps https://www.npmjs.com/package/query-string will work.